lib/ansible/modules/packaging/os/dnf.py

```diff
@@ -724,38 +724,31 @@ def _mark_package_install(self, pkg_spec, upgrade=False):
         is_newer_version_installed = self._is_newer_version_installed(pkg_spec)
         is_installed = self._is_installed(pkg_spec)
         try:
-            if self.allow_downgrade:
-                # dnf only does allow_downgrade, we have to handle this ourselves
-                # because it allows a possibility for non-idempotent transactions
-                # on a system's package set (pending the yum repo has many old
-                # NVRs indexed)
-                if upgrade:
-                    if is_installed:
-                        self.base.upgrade(pkg_spec)
+            if is_newer_version_installed:
+                if self.allow_downgrade:
+                    # TODO: Copied from existing code, no idea how this really works
```

Comment:
Why was this comment added? `# TODO: Copied from existing code, no idea how this really works`

Reply:
I do not understand the logic of this part of the if (This part is left as it was before my changes) -- hence the comment, would remove it before commit.

Reply:
I do not understand the logic of this part of the if (This part is left as it was before my changes) -- hence the comment, would remove it before commit but thought someone could explain to me how this part works :).